src/elements/dialog/dialog/dialog.spec.ts

```diff
@@ -495,4 +500,110 @@ describe('sbb-dialog', () => {
       expect(element).not.to.have.attribute('data-hide-header');
     });
   });
+
+  describe('with autocomplete', () => {
+    const openDialog: (root: HTMLElement, selector: string) => void = (
+      root: HTMLElement,
+      selector: string,
+    ) => {
+      (root.querySelector(selector) as SbbDialogElement)!.open();
+    };
+
+    it('pressing "Escape" will close only the last opened overlay', async () => {
+      const root: HTMLElement = await fixture(html`
+        <div>
+          <sbb-button id="button-1" @click=${() => openDialog(root, '#dialog-1')}
+            >Open dialog</sbb-button
+          >
+          <sbb-dialog id="dialog-1">
+            <sbb-dialog-title>Title</sbb-dialog-title>
+            <sbb-dialog-content>
+              <sbb-button id="button-2" @click="${() => openDialog(root, '#dialog-2')}"
+                >Open nested dialog</sbb-button
+              >
+              <sbb-dialog id="dialog-2">
+                <sbb-dialog-content>
+                  Content
+                  <sbb-form-field>
+                    <label>Pressing 'Escape' keydown with overlay</label>
+                    <input />
+                    <sbb-autocomplete>
+                      <sbb-option value="1">1</sbb-option>
+                      <sbb-option value="2">2</sbb-option>
+                      <sbb-option value="3">3</sbb-option>
+                    </sbb-autocomplete>
+                  </sbb-form-field>
+                </sbb-dialog-content>
+              </sbb-dialog>
+            </sbb-dialog-content>
+          </sbb-dialog>
+        </div>
+      `);
+
+      const button = root.querySelector<SbbButtonElement>('#button-1')!;
+      const dialog = root.querySelector<SbbDialogElement>('#dialog-1')!;
+      const nestedButton = root.querySelector<SbbButtonElement>('#button-2')!;
+      const nestedDialog = root.querySelector<SbbDialogElement>('#dialog-2')!;
+      const input = root.querySelector<HTMLInputElement>('input')!;
+      const autocomplete = root.querySelector<SbbAutocompleteElement>('sbb-autocomplete')!;
+
+      // the overlays are all closed
+      expect(dialog).to.have.attribute('data-state', 'closed');
+      expect(dialog).not.to.match(':popover-open');
+      expect(nestedDialog).to.have.attribute('data-state', 'closed');
+      expect(nestedDialog).not.to.match(':popover-open');
+      expect(input).to.have.attribute('aria-expanded', 'false');
+      expect(autocomplete).not.to.match(':popover-open');
```

Comment:
very nitpicky, I think asserting only the data-state is enough (you only want to be sure that you are in the right state)